library/Icingadb/Authentication/ObjectAuthorization.php

```diff
@@ -208,4 +216,37 @@ protected function checkGrants($permission, $roles)
 
         return false;
     }
+
+    /**
+     * Check whether the given filter matches the given object
+     *
+     * @param string $restriction
+     * @param Model  $object
+     *
+     * @return bool
+     */
+    public static function matchesOn($restriction, Model $object)
+    {
+        $self = new static();
+
+        $tableName = $object->getTableName();
+        $uniqueId = $object->{$object->getKeyName()};
+        if (! isset(self::$matchedFilters[$tableName][$uniqueId])) {
```

Comment:
Don't use the table name, use the restriction. There may be different filters for the same object type.